src/helpers/page/WrapSetTimeoutInPromise.ts

```diff
@@ -0,0 +1,13 @@
+
+export async function wrapSetTimeoutInPromise(callback: Function, timeDelaySeconds: number): Promise<void> {
```

Comment:
I don't understand why this takes in a callback and also returns a `Promise`.

It looks like we are trying to have a version of `setTimeout` we can use with `await`. If this is the case then this would be a simpler way to do this.
```typescript
function awaitableTimeout(ms: number) :Promise<void> {
    return new Promise(resolve => setTimeout(resolve, ms));
}
```

Reply:
The callback is for the actual call to show the native or slidedown prompt after the setTimeout. It's wrapped in a promise that is returned to the higher level method so that the entire `showDelayedPrompt` method can be awaited on. 

See https://github.com/OneSignal/OneSignal-Website-SDK/pull/630/files#diff-1d83af775103f5a84ed90068ae19c422R133